src/corva/cache_adapter.py

```diff
@@ -0,0 +1,258 @@
+from datetime import timedelta
+from typing import Dict, List, Optional, Protocol, Union
+
+import redis
+
+
+class CacheRepositoryProtocol(Protocol):
+    def set(
+        self,
+        key: str,
+        value: str,
+        ttl: int,
+    ) -> None:
+        ...
+
+    def get(self, key: str) -> Optional[str]:
+        ...
+
+    def delete(self, key: str) -> None:
+        ...
+
+
+class RedisRepository:
```

Comment:
I need to spend some time tomorrow to understand what these scripts do

Reply:
The [repository tests](https://github.com/corva-ai/python-sdk/blob/feature/dc-2396_cache-with-separate-key-expiry/tests/integration/test_redis_repository.py) may help to understand quicker